lib/Doctrine/Persistence/Reflection/RuntimePublicReflectionProperty.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Doctrine\Persistence\Reflection;
+
+use Doctrine\Common\Proxy\Proxy;
```

Comment:
So this is where the package depends on the common library. It doesn't require it but supports it (hence the test). Would it make sense to just make the test skip if the library is not loaded and add it as a suggestion?

Reply:
So to be clear, you would add `composer require --dev doctrine/common` to the CI script? In the end, the difference it would make would be that contributors would run part of the tests and would not have to download that lib, at the expense of slightly longer builds in our CI, so… I'm not really convinced but can do the change. Let's see if other contributors feel strongly one way or another?

Reply:
> So to be clear, you would add `composer require --dev doctrine/common` to the CI script?

Yes, this was the idea. Technically, the development of the script doesn't strictly depend on this package. But I see what you're saying.

Reply:
> How is it different from this case?

Here, it doesn't depend on extensions. Extensions are more involving to install: you have to be root, composer can't do it for you, and in some cases you might even not have a system package for them. That's why it makes a lot of sense to proceed like this on the DBAL. `doctrine/common` has 2 dependencies: this very lib, and php, so it means no extra dep. It's just not worth it IMO.

Reply:
I can live with `composer require --dev doctrine/common` or adding it to the dev dependencies. I've seen both ways in different projects.

Reply:
If I may summarize it, if there's no reason for the change, let's not make it.

Reply:
The other option is to copy the interface here and add a autoload-dev. Since its just a marker interface with no methods, there is no risk of breaking with or without doctrine/common.

Reply:
I added the requirement under the `require-dev` section, so it's not like there is an impact on the users.